presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnValidator.java

```diff
@@ -16,82 +16,205 @@
                 new SingleColumn(arrayCardinalityChecksum, Optional.of(delimitedIdentifier(getCardinalityChecksumColumnAlias(column)))),
                 new SingleColumn(arrayCardinalitySum, Optional.of(delimitedIdentifier(getCardinalitySumColumnAlias(column)))));
     }
 
-    @Override
-    public List<ColumnMatchResult<ArrayColumnChecksum>> validate(Column column, ChecksumResult controlResult, ChecksumResult testResult)
-    {
-        ArrayColumnChecksum controlChecksum = toColumnChecksum(column, controlResult);
-        ArrayColumnChecksum testChecksum = toColumnChecksum(column, testResult);
-
-        return ImmutableList.of(new ColumnMatchResult<>(Objects.equals(controlChecksum, testChecksum), column, controlChecksum, testChecksum));
-    }
-
     public static Expression generateArrayChecksum(Expression column, Type type)
     {
         checkArgument(type instanceof ArrayType, "Expect ArrayType, found %s", type.getDisplayName());
         Type elementType = ((ArrayType) type).getElementType();
 
         if (elementType.isOrderable()) {
-            FunctionCall arraySort = new FunctionCall(QualifiedName.of("array_sort"), ImmutableList.of(column));
+            Expression arraySort = functionCall("array_sort", column);
 
             if (elementType instanceof ArrayType || elementType instanceof RowType) {
                 return new CoalesceExpression(
-                        new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(new TryExpression(arraySort))),
-                        new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(column)));
+                        functionCall("checksum", new TryExpression(arraySort)),
+                        functionCall("checksum", column));
             }
-            return new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(arraySort));
+            return functionCall("checksum", arraySort);
+        }
+        return functionCall("checksum", column);
+    }
+
+    private static Expression generateInfinityLambdaExpression(ArithmeticUnaryExpression.Sign sign)
+    {
+        ComparisonExpression lambdaBody = new ComparisonExpression(
+                ComparisonExpression.Operator.EQUAL,
+                new Identifier("x"),
+                new ArithmeticUnaryExpression(sign, functionCall("infinity")));
+        return new LambdaExpression(ImmutableList.of(new LambdaArgumentDeclaration(identifier("x"))), lambdaBody);
+    }
+
+    private static Expression generateLambdaExpression(String functionName)
+    {
+        return new LambdaExpression(
+                ImmutableList.of(new LambdaArgumentDeclaration(identifier("x"))),
+                functionCall(functionName, new Identifier("x")));
+    }
+
+    @Override
+    public List<ColumnMatchResult<ArrayColumnChecksum>> validate(Column column, ChecksumResult controlResult, ChecksumResult testResult)
```

Comment:
NIT: Can we put this function to its original place? (before function generateArrayChecksum). we should put public functions ahead then private function

Reply:
I placed them like this of better see the logic of the code:
public generateChecksumColumns() uses the following generateArrayChecksum(), generateInfinityLambdaExpression() and generateLambdaExpression().

public validate() uses the following toColumnChecksum().

I can bring them back, no issue.

Reply:
Will be addressed in the next push.